client/src/Try/Container.purs

```diff
@@ -0,0 +1,469 @@
+module Try.Container where
+
+import Prelude
+
+import Ace (Annotation)
+import Control.Monad.Except (runExceptT)
+import Data.Array (fold)
+import Data.Array as Array
+import Data.Either (Either(..), hush)
+import Data.Foldable (for_)
+import Data.FoldableWithIndex (foldMapWithIndex)
+import Data.Maybe (Maybe(..), fromMaybe, isNothing)
+import Data.Symbol (SProxy(..))
+import Effect (Effect)
+import Effect.Aff (Aff)
+import Effect.Class.Console (error)
+import Effect.Uncurried (EffectFn1, runEffectFn1)
+import Foreign.Object (Object)
+import Foreign.Object as Object
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Try.API (CompileError(..), CompileResult(..), CompilerError, ErrorPosition)
+import Try.API as API
+import Try.Config as Config
+import Try.Editor (MarkerType(..), toStringMarkerType)
+import Try.Editor as Editor
+import Try.Gist (getGistById, tryLoadFileFromGist)
+import Try.Loader (Loader, makeLoader, runLoader)
+import Try.QueryString (getQueryStringMaybe)
+import Try.Session (createSessionIdIfNecessary, storeSession, tryRetrieveSession)
+import Try.Types (JS(..))
+import Web.HTML (window)
+import Web.HTML.Window (alert)
+
+type Slots = ( editor :: Editor.Slot Unit )
```

Comment:
I've chosen to implement this as a container component for the page and a dedicated editor component for just the code editor. It makes it a little easier to focus just on the editor when implementing new features for it, or just focus on the container when dealing with the full page. I'm open to different architectures, however.